source/world_builder/utilities.cc

```diff
@@ -1330,6 +1331,17 @@ namespace WorldBuilder
           const double spreading_velocity_point0 = mid_oceanic_spreading_velocities[relevant_ridge][i_coordinate];
           const double spreading_velocity_point1 = mid_oceanic_spreading_velocities[relevant_ridge][i_coordinate + 1];
 
+          double subducting_velocity_point0 = subducting_plate_velocities[0][0];
+          double subducting_velocity_point1 = subducting_plate_velocities[0][0];
+
+          ridge_migration_time = ridge_migration_times[relevant_ridge];
```

Comment:
This seems to be the location where valgrind actually trips over. I think it is caused by faulty input at the place where I posted the valgrind output, but it would be good to add asserts here (or at the start of this function) to check whether the input you are providing is valid.